services/src/main/java/org/keycloak/services/managers/ApplianceBootstrap.java

```diff
@@ -127,4 +139,25 @@ public void createMasterRealmUser(String username, String password) {
         ServicesLogger.LOGGER.addUserSuccess(username, Config.getAdminRealm());
     }
 
+    public void createTemporaryMasterRealmAdminService(String clientId, String clientSecret, Integer expriationMinutes) {
+        RealmModel realm = session.realms().getRealmByName(Config.getAdminRealm());
+        session.getContext().setRealm(realm);
+
+        clientId = StringUtil.isBlank(clientId) ? DEFAULT_TEMP_ADMIN_SERVICE : clientId;
+        expriationMinutes = expriationMinutes == null ? DEFAULT_TEMP_ADMIN_EXPIRATION : expriationMinutes;
+
+        ClientModel adminClient = session.clients().addClient(realm, clientId);
+        adminClient.setEnabled(true);
+        // TODO: set temporary
+        // also set the expiration - could be relative to a creation timestamp, or computed
+
+        adminClient.setSecret(clientSecret);
+
+        adminClient.addRole(AdminRoles.ADMIN);
```

Comment:
I believe this adds a new Client Role to the newly created Client.

What we need is to enable service account and assign Realm Admin role to it. Could you please fix it?

Reply:
Should be good now.